src/main/java/com/flightstats/hub/config/properties/LocalHostProperties.java

```diff
@@ -0,0 +1,73 @@
+package com.flightstats.hub.config.properties;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+@Slf4j
+public class LocalHostProperties {
+
+    private String uriScheme = "http://";
+    private final int port;
+
+    @Inject
+    public LocalHostProperties(AppProperties appProperties, SystemProperties systemProperties) {
+        this.port =  systemProperties.getHttpBindPort();
+        if (appProperties.isAppEncrypted()) {
```

Comment:
In order to avoid having logic in the constructor and a non-final variable above, I think I'd extract this logic into `getUriScheme` and add an `else` that captures the default value.